src/optimization/TrajoptUtil.hpp

```diff
@@ -349,28 +395,35 @@ inline Solution GenerateLinearInitialGuess(
           initialGuess.y,
           Linspace(initialGuessPoints.at(wptIdx).at(guessPointCount - 2).y,
                    initialGuessPoints.at(wptIdx).back().y, N_lastGuessSgmt));
-      append_vector(
-          initialGuess.theta,
-          Linspace(
-              initialGuessPoints.at(wptIdx).at(guessPointCount - 2).heading,
-              initialGuessPoints.at(wptIdx).back().heading, N_lastGuessSgmt));
+      auto lastTheta = AngleLinspace(
+          initialGuessPoints.at(wptIdx).at(guessPointCount - 2).heading,
+          initialGuessPoints.at(wptIdx).back().heading, N_lastGuessSgmt);
+      for (auto it = lastTheta.cbegin(); it != lastTheta.cend(); it++) {
+        initialGuess.thetacos.push_back(std::cos(*it));
+        initialGuess.thetasin.push_back(std::sin(*it));
```

Comment:
Use range-based for instead of iterators. There's instances of this kind of things earlier too.
```suggestion
      for (const auto& lastTheta : lastThetas) {
        initialGuess.thetacos.push_back(std::cos(lastTheta));
        initialGuess.thetasin.push_back(std::sin(lastTheta));
```